doc/classes/ProjectSettings.xml

```diff
@@ -530,12 +530,11 @@
 		<member name="display/window/size/width" type="int" setter="" getter="" default="1024">
 			Sets the game's main viewport width. On desktop platforms, this is the default window size. Stretch mode settings also use this as a reference when enabled.
 		</member>
-		<member name="display/window/vsync/use_vsync" type="bool" setter="" getter="" default="true">
-			If [code]true[/code], enables vertical synchronization. This eliminates tearing that may appear in moving scenes, at the cost of higher input latency and stuttering at lower framerates. If [code]false[/code], vertical synchronization will be disabled, however, many platforms will enforce it regardless (such as mobile platforms and HTML5).
-		</member>
-		<member name="display/window/vsync/vsync_via_compositor" type="bool" setter="" getter="" default="false">
-			If [code]Use Vsync[/code] is enabled and this setting is [code]true[/code], enables vertical synchronization via the operating system's window compositor when in windowed mode and the compositor is enabled. This will prevent stutter in certain situations. (Windows only.)
-			[b]Note:[/b] This option is experimental and meant to alleviate stutter experienced by some users. However, some users have experienced a Vsync framerate halving (e.g. from 60 FPS to 30 FPS) when using it.
+		<member name="display/window/vsync/vsync_mode" type="String" setter="" getter="" default="&quot;Enabled&quot;">
+			Sets the VSync mode for the main game window.
+			With [code]Enabled[/code] and [code]Mailbox[/code] the image is displayed only on vertical blanking periods, so no tearing is visible.
+			[code]Adaptive[/code] behaves like disabled VSync when the framerate drops below the screen's refresh rate to reduce stuttering(although tearing may be visible), otherwise VSync is enabled to avoid tearing.
```

Comment:
```suggestion
			[code]Adaptive[/code] behaves like [code]DISABLED[/code] VSync when the framerate drops below the screen's refresh rate to reduce stuttering(although tearing may be visible), otherwise VSync is enabled to avoid tearing.
```

Reply:
Linked the enum destription instead of describing the different modes directly, so that they are at one place.